### ISSUE TITLE
"lerna publish" does not remove the "workspace:" prefix

### ISSUE THREAD

**Issue by zepang** [lerna/lerna#3393]
## Current Behavior
There is a problem when I run the command "lerna publish", it does not remove the "workspace:" prefix

## Expected Behavior

The "lerna publish" command removes the "workspace:" prefix and turns it into a normal version
## Steps to Reproduce

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs / Configuration
The following is the output log when i install the package which published by then "lerna"
```shell
Couldn't find any versions for "@kso/jsencrypt" that matches "workspace:^3.0.725"
? Please choose a version of "@kso/jsencrypt" from this list: (Use arrow keys)
...
```

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
{
  "packages": ["packages/*"],
  "version": "independent",
  "npmClient": "pnpm",
  "useWorkspaces": true,
  "command": {
    "publish": {
      "ignoreChanges": ["*.md", "*.txt", "test/**"]
    }
  }
}
```

</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```

</p></details>

### Environment
System:
OS: macOS 12.5.1
CPU: (8) x64 Apple M1
Binaries:
Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
Yarn: 1.22.17 - ~/.nvm/versions/node/v14.19.0/bin/yarn
npm: 6.14.16 - ~/.nvm/versions/node/v14.19.0/bin/npm
Utilities:
Git: 2.31.0 - /usr/local/bin/git

**Comment by zqinmiao** [comment]
The lerna version I used was `6.0.1` and had the same problem.

**Comment by zepang** [comment]
@fahslaj Thank you for your reply, but this merge request is still being processed because the e2e test failed, I hope to fix this problem as soon as possible

**Comment by fahslaj** [comment]
Thanks @zepang . I have updated the e2e tests.

**Comment by JamesHenry** [comment]
This should hopefully be fully resolved in `6.0.3`
